src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -1161,35 +1158,49 @@ private ImmutableArray<BoundNode> ConvertCollectionExpressionElements(
                                 implicitReceiver,
                                 static (binder, syntax, item, implicitReceiver, arg, diagnostics) =>
                                 {
-                                    Debug.Assert(ConversionsBase.IsKeyValuePairType(binder.Compilation, item.Type, out _, out _));
+                                    BoundExpression convertedExpression;
+                                    if (arg.elementConversion.TryGetKeyValueConversions(out var keyConversion, out var valueConversion))
+                                    {
+                                        Debug.Assert(ConversionsBase.IsKeyValuePairType(binder.Compilation, item.Type, out _, out _));
+                                        convertedExpression = item;
```

Comment:
Should we also `BindToNaturalType` here?

Reply:
That shouldn't be necessary since `item` is an `BoundValuePlaceholder` with non-null type.

Reply:
Thanks
nit: could assert `!NeedsToBeConverted`